### PR TITLE
Bug 1740631 - [iscript] Add ability to specify requirements to productbuild command

### PR DESCRIPTION
**Pull request by ahal**

### CODE REVIEW

**Comment by ahal** [comment]
I'll need some help on how to test this, but it's probably best to get an initial review first.

File: iscript/src/iscript/mac.py [hunk 1]
```diff
@@ -1179,7 +1210,7 @@ async def notarize_behavior(config, task):
         # Unlock keychain again in case it's locked since previous unlock
         await unlock_keychain(sign_config["signing_keychain"], sign_config["keychain_password"])
         await update_keychain_search_path(config, sign_config["signing_keychain"])
-        await create_pkg_files(config, sign_config, all_paths)
+        await create_pkg_files(config, sign_config, all_paths, requirements_plist_path)
```
Comment (escapewindow) [inline iscript/src/iscript/mac.py]: I tend to prefer explicit `requirements_plist_path=requirements_plist_path` for kwargs, but a) i'm not sure if that's pythonic and b) it looks like we've already gone this route elsewhere. Non-blocking :)

**Review by escapewindow — APPROVED** [review]
Looks good, pending manual tests.
(I hadn't seen this review request when I suggested hneiva could look at this; gmta? :)

For testing, I'm thinking:

1. `./mach try empty` a try push, only has to include the requirements plist, doesn't have to have any taskgraph changes in it yet. get the url to the raw plist.

    If you want to include the taskgraph changes, we configure the entitlements urls [here](https://searchfox.org/mozilla-central/source/taskcluster/ci/config.yml#615) (schema [here](https://searchfox.org/mozilla-central/source/taskcluster/gecko_taskgraph/config.py#125)), the transforms are [here](https://searchfox.org/mozilla-central/source/taskcluster/gecko_taskgraph/transforms/signing.py#76) and [here](https://searchfox.org/mozilla-central/source/taskcluster/gecko_taskgraph/transforms/task.py#838)

- [quarantine](https://github.com/mozilla-releng/scriptworker-scripts/wiki/Testing-iscript#initial-setup) a prod mac signer, ssh in, update with your code

- follow [these steps](https://github.com/mozilla-releng/scriptworker-scripts/wiki/Testing-iscript#populate-the-work-dir-test) to download the task.json, download the upstreamArtifacts into the work_dir, edit the task.json to update the behavior to `mac_notarize` (this is so we do the whole thing: sign, submit for notarization, poll Apple, and staple the notarization; these are normally in 3 different tasks)

    -  (also update the task.json to add your requirements plist url -- you could test with and without this if wanted)

- don't forget to [clean up](https://github.com/mozilla-releng/scriptworker-scripts/wiki/Testing-iscript#finish-up) afterwards. if you didn't create a separate venv, delete your venv and /usr/local/bin/run_puppet.sh as root

LMK if you have questions!

**Comment by ahal** [comment]
Thanks! Fyi the Gecko patch is already written here:
https://phabricator.services.mozilla.com/D135056

I'll wait for this to land before pushing on it much more though.

**Comment by ahal** [comment]
Alright, I've tested this both with and without `requirements-plist-url` in the task.json, and there were no errors. So I think we're good to land and deploy here.

**Pull request merged**